crates/storage/provider/src/traits/transactions.rs

```diff
@@ -63,3 +63,25 @@ pub trait TransactionsProvider: BlockNumReader + Send + Sync {
     /// Returns None if the transaction is not found.
     fn transaction_sender(&self, id: TxNumber) -> RethResult<Option<Address>>;
 }
+
+///  Client trait for fetching additional [TransactionSigned] related data.
+#[auto_impl::auto_impl(&, Arc)]
+pub trait TransactionsProviderExt: BlockReader + Send + Sync {
+    /// Get transactions range by block range.
+    fn transaction_range_by_block_range(
```

Comment:
sick, I think many existing usages of `block_body_indices` can be replaced with that